### PR TITLE
LPS-183167 Changing a profile client-extension yaml file should make certain tasks out of date

### PR DESCRIPTION
**Pull request by drewbrokke**
This is an update for [LPS-183167](https://issues.liferay.com/browse/LPS-183167).

This pull fixes a number of issues with Gradle's incremental build when building/deploying client extensions. Some highlights:
- Fixes incorrectly annotated outputs in `CreateClientExtensionConfigTask`
- Conditionally include the `buildSiteInitializerZip` task rather than always including as a no-op
- Replaces lambdas with anonymous classes for `doLast`/`doFirst` statements so that Gradle's incremental build is not automatically invalidated. See [Gradle's docs on validation_problems.html#implementation_unknown](https://docs.gradle.org/7.3.3/userguide/validation_problems.html#implementation_unknown)
- Declares the validation tasks' outputs as always up-to-date since they don't produce any output
- Moves all declarations of `client-extension.yaml` as an input to a single location
- Conditionally declares the profile `yaml` files as inputs so that things will properly re-build when switching between different deploy profile tasks

FYI @gamerson @rotty3000 @petershin

### CODE REVIEW

File: modules/sdk/gradle-plugins-workspace/src/main/java/com/liferay/gradle/plugins/workspace/task/CreateClientExtensionConfigTask.java [hunk 1]
```diff
@@ -205,17 +206,17 @@ public File getInputPluginPackagePropertiesFile() {
 		return GradleUtil.toFile(_project, "liferay-plugin-package.properties");
 	}
 
-	@InputFiles
+	@OutputFile
```
Comment (gamerson) [inline modules/sdk/gradle-plugins-workspace/src/main/java/com/liferay/gradle/plugins/workspace/task/CreateClientExtensionConfigTask.java:209]: This is an input file right?  the `${project}/LCP.json`

File: modules/sdk/gradle-plugins-workspace/src/main/java/com/liferay/gradle/plugins/workspace/task/CreateClientExtensionConfigTask.java [hunk 2]
```diff
@@ -180,12 +181,12 @@ classificationGrouping, getInputLcpJsonFile(), getLcpJsonFile(),
 		_createClientExtensionConfigFile(jsonMap);
 	}
 
-	@InputFiles
+	@OutputFile
```
Comment (gamerson) [inline modules/sdk/gradle-plugins-workspace/src/main/java/com/liferay/gradle/plugins/workspace/task/CreateClientExtensionConfigTask.java:184]: 😆 oops

File: modules/sdk/gradle-plugins-workspace/src/main/java/com/liferay/gradle/plugins/workspace/task/CreateClientExtensionConfigTask.java (hunk 1 above)
Comment (drewbrokke) [reply modules/sdk/gradle-plugins-workspace/src/main/java/com/liferay/gradle/plugins/workspace/task/CreateClientExtensionConfigTask.java:209]: There should be a corresponding input method called `getInputLcpJsonFile`.

Comment (gamerson) [reply modules/sdk/gradle-plugins-workspace/src/main/java/com/liferay/gradle/plugins/workspace/task/CreateClientExtensionConfigTask.java:209]: oh i see the `_lcpJsonFile` is set by the in the constructor via the _addTaskOutputFile()

**Review by gamerson — APPROVED** [review]
LGTM!

**Comment by drewbrokke** [comment]
ci:forward

**Pull request closed without merging**